### PR TITLE
Add SaladCloud API Key detector

### PR DESCRIPTION
**Pull request by seniorquico**
### Description:

This adds an API key detector for SaladCloud (https://salad.com/). SaladCloud is a community/distributed cloud provider.

I also opened a "How To Rotate" guide PR: https://github.com/trufflesecurity/how-to-rotate/pull/29

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/saladcloudapikey/saladcloudapikey.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	apiKey        = regexp.MustCompile(`\b(salad_cloud_[0-9A-Za-z]{1,7}_[0-9A-Za-z]{7,235})\b`)
```
Comment (abmussani) [inline pkg/detectors/saladcloudapikey/saladcloudapikey.go:25]: @seniorquico Is there any reason to select the range of `{7,235}`? I generated three to four times the API Access key, All of the time, the last part, I got of length 49 characters.

File: pkg/detectors/saladcloudapikey/saladcloudapikey.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		results = append(results, s1)
+	}
+
+	return
+}
+
+func verifyMatch(ctx context.Context, client *http.Client, token string) (bool, map[string]string, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://api.salad.com/api/public", nil)
```
Comment (abmussani) [inline pkg/detectors/saladcloudapikey/saladcloudapikey.go:71]: @seniorquico Are you sure this is the correct url ?  I am also getting 404 with the correct Access key.

**Review by abmussani — COMMENTED** [review]
Thank you @seniorquico for the addition in detectors. I have posted few comments.

File: pkg/detectors/saladcloudapikey/saladcloudapikey.go (hunk 2 above)
Comment (seniorquico) [reply pkg/detectors/saladcloudapikey/saladcloudapikey.go:71]: @abmussani Apologies, it looked like there was a service issue earlier today. It has been resolved, and I am now successfully getting `204 No Content` and `401 Unauthorized` for good/bad API keys.

image
image

File: pkg/detectors/saladcloudapikey/saladcloudapikey.go (hunk 1 above)
Comment (seniorquico) [reply pkg/detectors/saladcloudapikey/saladcloudapikey.go:25]: We are currently wrapping up our GitHub Secret Scanning integration. We spec'd out the API keys similar to GitHub's based on this [blog post](https://github.blog/engineering/behind-githubs-new-authentication-token-formats/). The max 255 character length was just to keep options open for future scenarios with higher entropy keys.

I don't think it's a problem to shorten this if you think that's a better idea for the near-term. Please advise, and I can update the PR if necessary.

**Comment by seniorquico** [comment]
Rebased to resolve merge conflicts.

**Review by abasit-folio3 — APPROVED** [review]

**Review by abmussani — APPROVED** [review]

File: pkg/detectors/saladcloudapikey/saladcloudapikey.go (hunk 2 above)
Comment (abmussani) [reply pkg/detectors/saladcloudapikey/saladcloudapikey.go:71]: Thanks, endpoint is working now.

**Review by kashifkhan0771 — APPROVED** [review]

**Comment by zricethezav** [comment]
@seniorquico LGTM, would you mind resolving the conflicts? We'll get it merged in after they get resolved.

**Comment by seniorquico** [comment]
@zricethezav Rebased. Thanks!

**Review by ahrav — APPROVED** [review]

**Comment by seniorquico** [comment]
Thanks, @ahrav! Do you know if someone will be available to look at trufflesecurity/how-to-rotate#29 soon, as well? The changes in this PR included a link to that guide, but it's not yet live.

**Comment by zricethezav** [comment]
@seniorquico stand by, this change actually doesnt build since it doesnt satisfy the detector interface (missing description)

**Comment by seniorquico** [comment]
I'm sorry, @zricethezav! I just opened #3399.

**Pull request merged**